### PR TITLE
z-index sort

### LINKED ISSUE #120: weird outline around Text

### PR DESCRIPTION
**Pull request by x06lan**
- fix #120 by sort the render order of image
screenshot-2024-02-20-00-28-19
screenshot-2024-02-20-00-28-01

### CODE REVIEW

File: src/Util/Root.cpp [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #include "Util/Root.hpp"
 
 #include "Util/Logger.hpp"
+#include <queue>
```
Comment (NOOBDY) [inline src/Util/Root.cpp:4]:
```suggestion
#include <queue>

#include "Util/Logger.hpp"
```

File: src/Util/Root.cpp [hunk 2]
```diff
@@ -25,25 +26,39 @@ void Root::Update() {
     struct StackInfo {
         std::shared_ptr<GameObject> m_GameObject;
         Transform m_ParentTransform;
+        bool operator()(const StackInfo& lhs, const StackInfo& rhs) const {
+            return  lhs.m_GameObject->GetZIndex()> rhs.m_GameObject->GetZIndex();
+        }
```
Comment (NOOBDY) [inline src/Util/Root.cpp:31]: what is this for?

**Review by NOOBDY — CHANGES_REQUESTED** [review]
please run clang-tidy to format your code

**Comment by ntut-rick** [comment]
is this PR fixs #120 ?

**Review by ntut-rick — COMMENTED** [review]
BTW, I thought we had resolved the z-index sorting problem with OpenGL, didn't we?

File: src/Util/Root.cpp (hunk 2 above)
Comment (ntut-rick) [reply src/Util/Root.cpp:31]: wait this is `operator()`???

Comment (ntut-rick) [reply src/Util/Root.cpp:31]: SUSSY

File: src/Util/Root.cpp [hunk 3]
```diff
@@ -26,22 +26,21 @@ void Root::Update() {
...
         stack.push_back(StackInfo{child, Transform{}});
     }
 
-    std::priority_queue<StackInfo, std::vector<StackInfo>, decltype(compareFunction)> renderQueue(compareFunction);
+    auto compareFunction = [](const StackInfo &a, const StackInfo &b) {
+        return a.m_GameObject->GetZIndex() > b.m_GameObject->GetZIndex();
+    };
+    std::priority_queue<StackInfo, std::vector<StackInfo>,
+                        decltype(compareFunction)>
+        renderQueue(compareFunction);
```
Comment (ntut-rick) [inline src/Util/Root.cpp:43]: why this format looks so ugly  sadge

**Review by ntut-rick — DISMISSED** [review]
LGTM

**Review by NOOBDY — APPROVED** [review]

**Review by wj461 — APPROVED** [review]
IWOMM

**Pull request merged**